hack/generator/pkg/astmodel/one_of_json_marshal_function.go

```diff
@@ -122,3 +122,8 @@ func (f *OneOfJSONMarshalFunction) AsFunc(
 func (f *OneOfJSONMarshalFunction) RequiredPackageReferences() *PackageReferenceSet {
 	return NewPackageReferenceSet(MakeExternalPackageReference("encoding/json"))
 }
+
+// ReturnTypes returns nil as we can't represent byte[] in our type system
+func (f *OneOfJSONMarshalFunction) ReturnTypes() []Type {
+	return nil
```

Comment:
Presumably this isn't a problem because ReturnTypes is just used for References?

Reply:
I've removed this method - have added a new interface `ValueFunction` which is implemented only where needed, eliminating the need to add `ReturnTypes()` to a bunch of existing functions.